### PR TITLE
feat : Migrating registration verification fragment to MVVM

### PR DESCRIPTION
**Pull request by gururani-abhishek**
Fixes MOBILE-69

1. used RegistrationViewModel & UserAuthRepository to handle business logic and dataLayer logic for RegistrationVerificationFragment. 
2. moved userVerify payload creation to dataLayer(i.e. in UserAuthRepository)
3. covered registerationLogic(viewModel and repository) through proper unit testing -> using jUnit and Mockito
4. handled Ui state change of RegisterationVerification through RegistrationVerificaionUiState sealed class

https://github.com/openMF/mifos-mobile/assets/59947871/49bee274-b487-43e4-b7bf-518677e2ece1

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by gururani-abhishek** [comment]
Hey @jawidMuhammadi, please have a look at this whenever you find time :)

File: app/src/main/java/org/mifos/mobile/viewModels/RegistrationVerificationViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.observers.DisposableObserver
+import io.reactivex.schedulers.Schedulers
+import okhttp3.ResponseBody
+import org.mifos.mobile.api.DataManager
+import org.mifos.mobile.models.register.UserVerify
+import javax.inject.Inject
+
+class RegistrationVerificationViewModel @Inject constructor(private val dataManager : DataManager?) : ViewModel() {
```
Comment (jawidMuhammadi) [inline app/src/main/java/org/mifos/mobile/viewModels/RegistrationVerificationViewModel.kt]: No need to create this class, I believe you could use RegistrationViewModel instead and move the verification logic over there

**Review by jawidMuhammadi — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/mifos/mobile/viewModels/RegistrationVerificationViewModel.kt (hunk 1 above)
Comment (gururani-abhishek) [reply app/src/main/java/org/mifos/mobile/viewModels/RegistrationVerificationViewModel.kt]: alright, converting it to draft and then will update this after registrationFragment is done with migration.

Comment (gururani-abhishek) [reply app/src/main/java/org/mifos/mobile/viewModels/RegistrationVerificationViewModel.kt]: Done

**Comment by gururani-abhishek** [comment]
Hey @jawidMuhammadi, this is done from my side, resolved the conflicts as well. Do have a look whenever you find the bandwidth. 😃

**Review by jawidMuhammadi — APPROVED** [review]
LGTM! :)

**Pull request merged**